scene/animation/tween.cpp

```diff
@@ -375,264 +375,15 @@ Variant Tween::interpolate_variant(Variant p_initial_val, Variant p_delta_val, f
-			return r;
-		}
-
-		case Variant::COLOR: {
-			Color i = p_initial_val;
-			Color d = p_delta_val;
-			Color r;
-
-			APPLY_EQUATION(r);
-			APPLY_EQUATION(g);
-			APPLY_EQUATION(b);
-			APPLY_EQUATION(a);
-			return r;
-		}
-
-		default: {
-			return p_initial_val;
-		}
-	};
-#undef APPLY_EQUATION
-}
-
-Variant Tween::calculate_delta_value(Variant p_intial_val, Variant p_final_val) {
-	ERR_FAIL_COND_V_MSG(p_intial_val.get_type() != p_final_val.get_type(), p_intial_val, "Type mismatch between initial and final value: " + Variant::get_type_name(p_intial_val.get_type()) + " and " + Variant::get_type_name(p_final_val.get_type()));
-
-	switch (p_intial_val.get_type()) {
-		case Variant::BOOL: {
-			return (int)p_final_val - (int)p_intial_val;
-		}
-
-		case Variant::RECT2: {
-			Rect2 i = p_intial_val;
-			Rect2 f = p_final_val;
-			return Rect2(f.position - i.position, f.size - i.size);
-		}
-
-		case Variant::RECT2I: {
-			Rect2i i = p_intial_val;
-			Rect2i f = p_final_val;
-			return Rect2i(f.position - i.position, f.size - i.size);
-		}
-
-		case Variant::TRANSFORM2D: {
-			Transform2D i = p_intial_val;
-			Transform2D f = p_final_val;
-			return Transform2D(f.columns[0][0] - i.columns[0][0],
-					f.columns[0][1] - i.columns[0][1],
-					f.columns[1][0] - i.columns[1][0],
-					f.columns[1][1] - i.columns[1][1],
-					f.columns[2][0] - i.columns[2][0],
-					f.columns[2][1] - i.columns[2][1]);
-		}
-
-		case Variant::AABB: {
-			AABB i = p_intial_val;
-			AABB f = p_final_val;
-			return AABB(f.position - i.position, f.size - i.size);
-		}
-
-		case Variant::BASIS: {
-			Basis i = p_intial_val;
-			Basis f = p_final_val;
-			return Basis(f.rows[0][0] - i.rows[0][0],
-					f.rows[0][1] - i.rows[0][1],
-					f.rows[0][2] - i.rows[0][2],
-					f.rows[1][0] - i.rows[1][0],
-					f.rows[1][1] - i.rows[1][1],
-					f.rows[1][2] - i.rows[1][2],
-					f.rows[2][0] - i.rows[2][0],
-					f.rows[2][1] - i.rows[2][1],
-					f.rows[2][2] - i.rows[2][2]);
-		}
-
-		case Variant::TRANSFORM3D: {
-			Transform3D i = p_intial_val;
-			Transform3D f = p_final_val;
-			return Transform3D(f.basis.rows[0][0] - i.basis.rows[0][0],
-					f.basis.rows[0][1] - i.basis.rows[0][1],
-					f.basis.rows[0][2] - i.basis.rows[0][2],
-					f.basis.rows[1][0] - i.basis.rows[1][0],
-					f.basis.rows[1][1] - i.basis.rows[1][1],
-					f.basis.rows[1][2] - i.basis.rows[1][2],
-					f.basis.rows[2][0] - i.basis.rows[2][0],
-					f.basis.rows[2][1] - i.basis.rows[2][1],
-					f.basis.rows[2][2] - i.basis.rows[2][2],
-					f.origin.x - i.origin.x,
-					f.origin.y - i.origin.y,
-					f.origin.z - i.origin.z);
-		}
+	// Special case for bool.
+	if (p_initial_val.get_type() == Variant::BOOL) {
+		return run_equation(p_trans, p_ease, p_time, p_initial_val, p_delta_val, p_duration) >= 0.5;
+	}
 
-		default: {
-			return Variant::evaluate(Variant::OP_SUBTRACT, p_final_val, p_intial_val);
-		}
-	};
+	Variant ret;
+	Variant::add_animationwise(p_initial_val, p_delta_val, ret);
```

Comment:
The value of `ret` is constant during Tweener's lifetime, so we could look into caching it. But this requires changing the `interpolate_variant()` to take final value instead of delta.

Probably fine as is for now.